### PR TITLE
build(extend-php-support): extended PHP versions support to 8.x

### LINKED ISSUE #36: PHP8.2
Can we get PHP8.2 support? Thanks!

### PR DESCRIPTION
**Pull request by Mohammad-Haris**
Closes #36

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -2,9 +2,7 @@ name: Tests
 on:
   workflow_dispatch:
   push:
-    branches: [ v2-master ]
-  pull_request:
-    branches: [ v2-master ]
+    branches: [ "36-php-8.2-support" ]
```
Comment (asadali214) [inline .github/workflows/php.yml]: Kindly revert this change, it was only intended for testing, not for release

File: composer.json [hunk 2]
```diff
@@ -26,14 +26,14 @@
     }
   ],
   "require": {
-    "php": ">=7.2 <8.2",
+    "php": ">=7.2 <8.3",
     "ext-curl": "*",
     "ext-json": "*",
-    "apimatic/core-interfaces": "^0.1.0"
+    "apimatic/core-interfaces": "dev-3-php-8.2-support"
```
Comment (asadali214) [inline composer.json]: This change was also intended for testing. Please use the correct core-interfaces version that we are going to release after PHP 8.2 support

File: composer.lock [hunk 3]
```diff
@@ -4,24 +4,24 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
         "This file is @generated automatically"
     ],
-    "content-hash": "d59e38d08c48e1a49e11c9de95d6d9ed",
+    "content-hash": "a936e737473b60adca280b2a50fecda7",
     "packages": [
         {
             "name": "apimatic/core-interfaces",
-            "version": "0.1.0",
+            "version": "dev-3-php-8.2-support",
```
Comment (asadali214) [inline composer.lock]: run composer update locally after releasing core-interfaces, to update composer lock file in this PR

Comment (Mohammad-Haris) [reply composer.lock]: PHP 8.2 update branch is not merged in core-interfaces-php. Composer won't resolve.

**Review by asadali214 — COMMENTED** [review]
Changes looks good to me, will be approved after releasing core-interfaces

**Review by asadali214 — APPROVED** [review]
Approved
